docs/migration/v8-to-v9.md

```diff
@@ -0,0 +1,159 @@
+# Upgrading from 8.x to 9.x
```

Comment:
You might ask yourself what is this file doing here.

Well, I started off this PR with actually doing the full changes for v9, leading me to move the v8 migration guide into its separate file. This is actually quite a bit of work with changing the links and all that stuff.

For this reason I would actually keep this file here for when we do the official switch because I don't wanna throw away that work.